src/utils/todo/task-filing/index.ts

```diff
@@ -7,146 +7,133 @@ import { filingTypeToName } from './helper'
 
 /** Build TodoItemI from filing TaskToDoI  */
 // https://docs.google.com/spreadsheets/d/1rJY3zsrdHS2qii5xb7hq1gt-D55NsakJtdu9ld9d80U/edit?gid=0#gid=0
-export const buildFilingTodo = async (task: TaskI) : Promise<TodoItemI | null> => {
-  const filing = task.task.filing
-  const header = filing.header
-  const business = useBcrosBusiness()
+export const buildFilingTodo = async (task: TaskI) : Promise<TodoItemI> => {
+  const { bootstrapFiling, bootstrapLegalType, linkedNr } = useBcrosBusinessBootstrap()
+  const { currentBusiness } = useBcrosBusiness()
+  if (!!currentBusiness && !!bootstrapFiling) {
+    console.error('Attempted buildFilingTodo without initializing the business or bootstrap first.')
+    return
+  }
 
-  let newTodo: TodoItemI | null = null
+  const filing = task?.task?.filing
+  const header = filing?.header
+  const filingData = filing[header?.name]
+  if (!filing || !header || !filingData) {
+    console.error('ERROR - invalid task provided =', task)
+  }
 
-  let corpFullDescription = ''
-  if (business) {
-    corpFullDescription = GetCorpFullDescription(business.currentBusiness.legalType)
+  const corpFullDescription = GetCorpFullDescription(currentBusiness?.legalType || bootstrapLegalType)
+  const isFilingType = (filingTypes: FilingTypes[]) => filingTypes.includes(header.name)
+
+  const paymentStatusCode = header.paymentStatusCode
+  const payErrorObj = paymentStatusCode && await getPayErrorObj(paymentStatusCode)
+
+  const newTodo: TodoItemI = {
+    uiUuid: UUIDv4(),
+    name: header.name,
+    filingId: header.filingId,
+    title: getTitle(filing, corpFullDescription),
+    draftTitle: getDraftTitle(filing),
+    status: header.status,
+    enabled: task.enabled,
+    order: task.order,
+    paymentMethod: header.paymentMethod || null,
+    paymentToken: header.paymentToken || null,
+    payErrorObj,
+    isPayCompleted: (paymentStatusCode === 'COMPLETED')
   }
-  const agmExtension = filing.agmExtension
-  const agmLocationChange = filing.agmLocationChange
-  const alteration = filing.alteration
-  const amalgamation = filing.amalgamationApplication
-  const annualReport = filing.annualReport
-  const consentContinuationOut = filing.consentContinuationOut
-  const continuationOut = filing.continuationOut
-  const continuationIn = filing.continuationIn
-  const conversion = filing.conversion
-  const correction = filing.correction
-  const dissolution = filing.dissolution
-  const incorporationApplication = filing.incorporationApplication
-  const registration = filing.registration
-  const restoration = filing.restoration
-  const specialResolution = filing.specialResolution
-
-  if (header) {
```

Comment:
removed 'if' as we know the data exists that we need already. Most code below is the same. The 'if' blocks check the filing type instead of the different objs and a block was added / edited for bootstraps